group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -1187,63 +1230,62 @@ public void scheduleLoadOperation(
     ) {
         throwIfNotRunning();
         log.info("Scheduling loading of metadata from {} with epoch {}", tp, partitionEpoch);
+
         // Touch the state to make the runtime immediately aware of the new coordinator.
-        getOrCreateContext(tp);
+        maybeCreateContext(tp);
 
         scheduleInternalOperation("Load(tp=" + tp + ", epoch=" + partitionEpoch + ")", tp, () -> {
-            CoordinatorContext context = getOrCreateContext(tp);
-
-            if (context.state == CoordinatorState.FAILED) {
-                // When the coordinator has failed, we create a new context instead of
```

Comment:
How have we addressed this? Are we clearing the timeline data structures?

Reply:
I guess we clear it here (when loading):
```
 snapshotRegistry = new SnapshotRegistry(logContext);
 lastWrittenOffset = 0L;
 lastCommittedOffset = 0L;
 ```
 I guess my only other question was the motivation for this change. Did it turn out to be easier to reuse rather than create new after all?

Reply:
Correct. I found out that when using the lock in the context, it is actually easier/better to reuse vs to recreate it because when you delete it, you loose the lock.